frontend/javascripts/components/select_experience_domain.tsx

```diff
@@ -11,92 +12,79 @@ type Props = {
-    });
-  }
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const existingDomains = await getExistingExperienceDomains();
+      setDomains(existingDomains);
+    };
+    void fetchData();
+  }, []);
 
-  getUnusedDomains(): ExperienceDomainList {
-    return this.state.domains.filter((domain) => !this.props.alreadyUsedDomains.includes(domain));
-  }
+  const getUnusedDomains = (): ExperienceDomainList => {
+    return domains.filter((domain) => !alreadyUsedDomains.includes(domain));
+  };
 
-  onSearch = (domain: string) => {
-    this.setState({
-      currentlyEnteredDomain: domain,
-    });
+  const onSearch = (domain: string) => {
+    setCurrentlyEnteredDomain(domain);
   };
 
-  render() {
-    const {
-      value,
-      notFoundContent,
-      width,
-      disabled,
-      placeholder,
-      onSelect,
-      onChange,
-      allowCreation,
-    } = this.props;
-    const { currentlyEnteredDomain } = this.state;
-    let options = this.getUnusedDomains();
+  let options = getUnusedDomains();
 
-    if (
-      allowCreation &&
-      !options.includes(currentlyEnteredDomain) &&
-      currentlyEnteredDomain.trim() !== ""
-    ) {
-      options = [...options, currentlyEnteredDomain];
-    }
+  if (
+    allowCreation &&
+    !options.includes(currentlyEnteredDomain) &&
+    currentlyEnteredDomain.trim() !== ""
+  ) {
+    options = [...options, currentlyEnteredDomain];
+  }
 
-    return (
-      <Tooltip
-        placement="top"
-        title="Select an existing experience domain or create a new one by typing its name in this input field."
+  return (
+    <Tooltip
+      placement="top"
+      title="Select an existing experience domain or create a new one by typing its name in this input field."
+    >
+      <Select
+        showSearch
+        value={value}
+        optionFilterProp="children"
+        notFoundContent={notFoundContent}
+        style={{
+          width: `${width}%`,
+        }}
+        disabled={disabled}
+        placeholder={placeholder}
+        onSelect={onSelect}
+        onChange={onChange}
+        onSearch={onSearch}
       >
-        <Select
-          showSearch
-          value={value}
-          optionFilterProp="children"
-          notFoundContent={notFoundContent}
-          style={{
-            width: `${width}%`,
-          }}
-          disabled={disabled}
-          placeholder={placeholder}
-          onSelect={onSelect}
-          onChange={onChange}
-          onSearch={this.onSearch}
-        >
-          {options.map((domain) => (
-            <Select.Option key={domain}>{domain}</Select.Option>
-          ))}
-        </Select>
-      </Tooltip>
-    );
-  }
-}
+        {options.map((domain) => (
+          <Select.Option key={domain}>{domain}</Select.Option>
+        ))}
+      </Select>
+    </Tooltip>
+  );
+};
+
+SelectExperienceDomain.defaultProps = {
+  alreadyUsedDomains: [],
+  allowCreation: false,
+};
```

Comment:
this cna be removed. According to the props `allowCreation` is required and the optional `alreadyUsedDomains` already has a default value